ivy/functional/ivy/experimental/linear_algebra.py

```diff
@@ -185,24 +182,30 @@ def diagflat(
 
     Parameters
     ----------
-    x
-        Input data, which is flattened and set as the k-th diagonal of the output.
-    k
-        Diagonal to set.
-        Positive value means superdiagonal,
-        0 refers to the main diagonal,
-        and negative value means subdiagonal.
-    out
-        optional output array, for writing the result to. It must have a shape that the
-        inputs broadcast to.
+    x : Union[ivy.Array, ivy.NativeArray]
```

Comment:
I dont think there is a need to specify type-hints in the docstring itself. The function's signature does provide type-hints which is sufficient 🙂